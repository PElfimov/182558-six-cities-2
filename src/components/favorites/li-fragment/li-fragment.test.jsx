import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import toJSON from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({adapter: new Adapter()});

import LiFragment from './li-fragment';

const offers = [
  {
    id: 1,
    isFavorite: true,
    city: {
      name: `Biysk`,
    }
  },
  {
    id: 2,
    isFavorite: true,
    city: {
      name: `Paris`,
    }
  },
  {
    id: 3,
    isFavorite: true,
    city: {
      name: `Paris`,
    }
  },
  {
    id: 4,
    isFavorite: true,
    city: {
      name: `Paris`,
    }
  },
];

it(`LiFragment correctly render`, () => {
  const tree = shallow(
      <LiFragment
        offers={offers}
        city = {`Paris`}
      />);

  expect(toJSON(tree)).toMatchSnapshot();
});
