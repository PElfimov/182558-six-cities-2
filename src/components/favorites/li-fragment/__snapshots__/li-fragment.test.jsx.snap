// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LiFragment correctly render 1`] = `
<li
  className="favorites__locations-items"
>
  <div
    className="favorites__locations locations locations--current"
  >
    <div
      className="locations__item"
    >
      <Link
        className="locations__item-link"
        to="/Paris"
      >
        <span>
          Paris
        </span>
      </Link>
    </div>
  </div>
  <div
    className="favorites__places"
  >
    <HotelCard
      key="1"
      offer={
        Object {
          "city": Object {
            "name": "Biysk",
          },
          "id": 1,
          "isFavorite": true,
        }
      }
      onClick={[Function]}
      onHover={[Function]}
      type="favorites"
    />
    <HotelCard
      key="2"
      offer={
        Object {
          "city": Object {
            "name": "Paris",
          },
          "id": 2,
          "isFavorite": true,
        }
      }
      onClick={[Function]}
      onHover={[Function]}
      type="favorites"
    />
    <HotelCard
      key="3"
      offer={
        Object {
          "city": Object {
            "name": "Paris",
          },
          "id": 3,
          "isFavorite": true,
        }
      }
      onClick={[Function]}
      onHover={[Function]}
      type="favorites"
    />
    <HotelCard
      key="4"
      offer={
        Object {
          "city": Object {
            "name": "Paris",
          },
          "id": 4,
          "isFavorite": true,
        }
      }
      onClick={[Function]}
      onHover={[Function]}
      type="favorites"
    />
  </div>
</li>
`;
